### PR TITLE
Added Token Validation for GoogleSSO

### PR DESCRIPTION
**Pull request by mathieutrudeau**
- Added a step where Google Tokens are validated in the backend.
- When a user first signs in using their google account, their username will be generated using their email.
- Fixed an issue that caused the jwt-token cookie to not be included in the httpcontext. Which caused the signed in user to be signed out.

### CODE REVIEW

**Review by sFranchetto — APPROVED** [review]
The changes demonstrate a proactive approach to improving the authentication flow, addressing potential security concerns, and providing a smoother user experience. The commit messages are clear and concise, making it easy to understand the purpose of each change. The inclusion of these enhancements significantly contributes to the overall robustness and reliability of the authentication system. 

Well done on identifying and resolving these critical issues.

File: frontend/components/shared/Navbar.tsx [hunk 1]
```diff
@@ -53,7 +55,25 @@ export default function Navbar() {
...
+      sub: string,
+      id_token: string,
+      name: string,
+      picture:string
+    }
+  }
+
   useEffect(() => {
+
+    if (session !== null) {
```
Comment (BigSaladMan) [inline frontend/components/shared/Navbar.tsx:70]: Not sure if we want this in here chief

File: frontend/helpers/AuthHelper.tsx [hunk 2]
```diff
@@ -188,25 +188,53 @@ export default class AuthHelper{
...
+            withCredentials: true
+        }
 
-    public static loginGoogleSSO = async (email:string,username:string,sub:string,avatar:string): Promise<boolean> =>
-    {
-        // Send the request.
-        const res = await RestHelper.authGoogleSSORequest(email,username,sub,avatar);
+        try {
+            console.debug("Sending the following authGoogleSSORequest...");
+            console.debug(options);
```
Comment (BigSaladMan) [inline frontend/helpers/AuthHelper.tsx:213]: Same with this

**Review by BigSaladMan — APPROVED** [review]
remove a few of the console logs, then should be good to merge

File: frontend/pages/api/auth/[...nextauth].ts [hunk 3]
```diff
@@ -12,33 +13,51 @@ export default NextAuth({
...
-    await AuthHelper.loginGoogleSSO(profile.email,profile.name,profile.sub,(profile as any).picture);
+    const googleSSORequest: GoogleSSORequest = {
+       email: profile.email,
+       sub: profile.sub,
+       token: token,
+       avatar: (profile as any).picture,
+       name: profile.name
+     } 
+    const response = await AuthHelper.loginGoogleSSO(googleSSORequest);
+    console.log("Response Object:", response);
```
[the remark this answers is not in the document]
Comment (mathieutrudeau) [reply frontend/pages/api/auth/[...nextauth].ts:62]: Thats logged on the server

File: frontend/helpers/AuthHelper.tsx (hunk 2 above)
Comment (mathieutrudeau) [reply frontend/helpers/AuthHelper.tsx]: Done

**Pull request merged**